File: clients/android/app/src/main/kotlin/com/lextures/android/features/navigation/CourseDrawer.kt
package com.lextures.android.features.navigation

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material.icons.filled.GridView
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.lextures.android.R
import com.lextures.android.core.design.sceneBackground
import com.lextures.android.core.design.textPrimary
import com.lextures.android.core.i18n.L
import com.lextures.android.core.navigation.MobileDestinations
import com.lextures.android.features.home.HomeShellState

/**
 * Course-scoped navigation drawer: the first-level menu shown while inside a course.
 * Mirrors the web course sidebar with a Back affordance (→ global menu) and a Dashboard
 * shortcut (→ leave course).
 */
@Composable
fun CourseDrawer(shell: HomeShellState, modifier: Modifier = Modifier) {
    val groups = MobileDestinations.courseDrawerGroups(shell.activeCourseSections)
    Column(
        modifier = modifier
            .fillMaxSize()
            .background(sceneBackground()),
    ) {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(start = 16.dp, end = 16.dp, top = 16.dp, bottom = 12.dp),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(10.dp),
        ) {
            Icon(
                painter = painterResource(R.drawable.launch_logo),
                contentDescription = null,
                tint = Color.Unspecified,
                modifier = Modifier.size(32.dp),
            )
            Text(
                text = shell.activeCourse?.displayTitle ?: "Lextures",
                fontSize = 17.sp,
                fontWeight = FontWeight.SemiBold,
                color = textPrimary(),
                maxLines = 2,
            )
        }

        Column(modifier = Modifier.padding(horizontal = 10.dp)) {
            DrawerRow(
                label = L.text(R.string.mobile_drawer_back),
                icon = Icons.AutoMirrored.Filled.ArrowBack,
                selected = false,
                onClick = { shell.openGlobalDrawer() },
            )
            DrawerRow(
                label = L.text(R.string.mobile_drawer_dashboard),
                icon = Icons.Filled.GridView,
                selected = false,
                onClick = { shell.exitCourseToDashboard() },
            )
        }

        Column(
            modifier = Modifier
                .fillMaxWidth()
                .verticalScroll(rememberScrollState())
                .padding(horizontal = 10.dp, vertical = 4.dp),
        ) {
            groups.forEach { group ->
                DrawerGroupHeader(drawerString(group.titleRes))
                group.sections.forEach { section ->
                    DrawerRow(
                        label = L.text(courseSectionLabelRes(section)),
                        icon = courseSectionIcon(section),
                        selected = shell.activeCourseSection == section,
                        onClick = {
                            shell.activeCourseSection = section
                            shell.closeDrawer()
                        },
                    )
                }
            }
        }
    }
}

File: clients/ios/Lextures/Features/Navigation/CourseDrawer.swift
import SwiftUI

/// Course-scoped navigation drawer: the first-level menu shown while inside a course.
/// Mirrors the web course sidebar (Content / Collaboration / Grades / People / Manage)
/// with a Back affordance (→ global menu) and a Dashboard shortcut (→ leave course).
struct CourseDrawer: View {
    @Environment(AppShellModel.self) private var shell
    @Environment(\.colorScheme) private var colorScheme

    private var groups: [CourseDrawerGroup] {
        MobileDestinations.courseDrawerGroups(shell.activeCourseSections)
    }

    var body: some View {
        VStack(alignment: .leading, spacing: 0) {
            header

            VStack(spacing: 4) {
                DrawerRow(
                    label: L.text("mobile.drawer.back"),
                    systemImage: "chevron.left",
                    selected: false
                ) {
                    shell.openGlobalDrawer()
                }
                DrawerRow(
                    label: L.text("mobile.drawer.dashboard"),
                    systemImage: "square.grid.2x2.fill",
                    selected: false
                ) {
                    shell.exitCourseToDashboard()
                }
            }
            .padding(.horizontal, 10)

            ScrollView {
                VStack(alignment: .leading, spacing: 4) {
                    ForEach(groups) { group in
                        DrawerGroupHeader(title: group.title)
                        ForEach(group.sections, id: \.self) { section in
                            DrawerRow(
                                label: section.label,
                                systemImage: courseSectionIcon(section),
                                selected: shell.activeCourseSection == section
                            ) {
                                shell.activeCourseSection = section
                                shell.closeDrawer()
                            }
                        }
                    }
                }
                .padding(.horizontal, 10)
                .padding(.bottom, 24)
            }
        }
        .frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .topLeading)
        .background(LexturesTheme.sceneBackground(for: colorScheme).ignoresSafeArea())
    }

    private var header: some View {
        HStack(spacing: 10) {
            BrandLogoView(maxHeight: 30)
                .frame(width: 34, height: 34)
            Text(shell.activeCourse?.displayTitle ?? "Lextures")
                .font(LexturesTheme.displayFont(18))
                .lineLimit(2)
                .foregroundStyle(LexturesTheme.textPrimary(for: colorScheme))
            Spacer(minLength: 0)
        }
        .padding(.horizontal, 16)
        .padding(.top, 16)
        .padding(.bottom, 12)
    }
}

/// SF Symbol for each course workspace section (drawer rows).
func courseSectionIcon(_ section: CourseWorkspaceSection) -> String {
    switch section {
    case .overview: return "doc.text"
    case .modules: return "square.stack.3d.up"
    case .files: return "folder"
    case .library: return "books.vertical"
    case .discussions: return "bubble.left.and.bubble.right"
    case .feed: return "text.bubble"
    case .live: return "dot.radiowaves.left.and.right"
    case .officeHours: return "clock"
    case .grades: return "list.clipboard"
    case .mastery: return "chart.bar"
    case .people: return "person.2"
    case .grading: return "checkmark.rectangle.stack"
    case .attendance: return "calendar.badge.checkmark"
    case .evaluations: return "star.bubble"
    }
}
